src/test/groovy/pl/allegro/tech/build/axion/release/infrastructure/git/GitRepositoryTest.groovy

```diff
@@ -220,6 +219,22 @@ class GitRepositoryTest extends Specification {
         remoteRawRepository.tag.list()*.fullName == ['refs/tags/release-push']
     }
 
+    def "should not push commits if the pushTagsOnly flag is set to true"() {
+        given:
+        initializationOptions = ScmInitializationOptions.fromProject(project, 'origin', true)
+        repository = new GitRepository(project.file('./repo'), identity, initializationOptions)
```

Comment:
hm, this creates nested repository, i would be careful with that - maybe just create new Project object? (this is a bit of hack for creating temp file, but it also makes the domain look nicer :P)